File: core/src/main/java/com/sequenceiq/cloudbreak/controller/AuthenticatedUserService.java
package com.sequenceiq.cloudbreak.controller;

import com.sequenceiq.cloudbreak.common.model.user.IdentityUser;
import com.sequenceiq.cloudbreak.common.service.user.UserFilterField;
import com.sequenceiq.cloudbreak.service.user.UserDetailsService;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.oauth2.provider.OAuth2Authentication;
import org.springframework.stereotype.Service;

import javax.inject.Inject;

@Service
public class AuthenticatedUserService {

    @Inject
    private UserDetailsService userDetailsService;

    public IdentityUser getCbUser() {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        if (authentication instanceof OAuth2Authentication) {
            OAuth2Authentication oauth = (OAuth2Authentication) authentication;
            if (oauth.getUserAuthentication() != null) {
                String username = (String) authentication.getPrincipal();
                return userDetailsService.getDetails(username, UserFilterField.USERNAME);
            }
        }
        return null;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/AuthenticatedUserService.kt
package com.sequenceiq.cloudbreak.controller

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.provider.OAuth2Authentication
import org.springframework.stereotype.Service

import com.sequenceiq.cloudbreak.domain.CbUser
import com.sequenceiq.cloudbreak.service.user.UserDetailsService
import com.sequenceiq.cloudbreak.service.user.UserFilterField

@Service
class AuthenticatedUserService {

    @Autowired
    private val userDetailsService: UserDetailsService? = null

    val cbUser: CbUser?
        get() {
            val authentication = SecurityContextHolder.getContext().authentication
            if (authentication != null) {
                val oauth = authentication as OAuth2Authentication?
                if (oauth.getUserAuthentication() != null) {
                    val username = authentication.principal as String
                    return userDetailsService!!.getDetails(username, UserFilterField.USERNAME)
                }
            }
            return null
        }
}
